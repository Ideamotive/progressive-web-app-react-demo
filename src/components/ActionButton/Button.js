import React from 'react';
import styled from 'styled-components';
import { lighten } from 'polished';

export default styled.button`
    width: ${props => (props.width || '100px')};
    height: ${props => (props.height || '100px')};
    background-color: ${props => (
      (props.primary && props.theme.colors.primary) ||
      (props.secondary && props.theme.colors.secondary) ||
      (props.warning && props.theme.colors.warning) ||
      (props.danger && props.theme.colors.danger) ||
      (props.success && props.theme.colors.success) ||
      (props.info && props.theme.colors.info) ||
      '#ccc'
    )};
    border-radius:  ${props => ((props.height / 2) || '50px')};
    border-style: solid;
    border-width: 2px;
    border-color: ${props => (
      (props.primary && lighten(.4, props.theme.colors.primary)) ||
      (props.secondary && lighten(.4, props.theme.colors.secondary)) ||
      (props.warning && lighten(.4, props.theme.colors.warning)) ||
      (props.danger && lighten(.4, props.theme.colors.danger)) ||
      (props.success && lighten(.4, props.theme.colors.success)) ||
      (props.info && lighten(.4, props.theme.colors.info)) ||
      '#ccc'
    )};
    text-align: center;
    cursor: pointer;
    transition: transform .5s;
    will-change: transform;
    &:active {
       transform: scale(1.05);
    }
    &:hover{
       transform: scale(1.025);
    }
    &:focus{
      outline: none;
    }
  `;