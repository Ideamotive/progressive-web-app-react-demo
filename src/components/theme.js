
export default {
  colors: {
    primary: '#DF691A',
    secondary: '#4E5D6C',
    warning: '#f0ad4e',
    danger: '#d9534f',
    success: '#5cb85c',
    info: '#5bc0de',
  }
}