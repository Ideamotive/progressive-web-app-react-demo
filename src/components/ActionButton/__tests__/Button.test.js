import Button from '../Button';
import React from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeProvider, } from 'styled-components';
import theme from '../../../components/theme';

describe('<Button/> component', () => {
  test('should match to snapshot', () => {
    const tree = renderer
      .create(<Button/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should modification match to snapshot', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <React.Fragment>
            <Button primary/>
            <Button secondary/>
            <Button warning/>
            <Button danger/>
            <Button success/>
            <Button info/>
          </React.Fragment>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test('should render ', () => {
  //   const iconClass = 'test';
  //   const wrapper = mount(<Button />);
  //   expect(wrapper.find(`i.${iconClass}`)).toHaveLength(1)
  // });

});

