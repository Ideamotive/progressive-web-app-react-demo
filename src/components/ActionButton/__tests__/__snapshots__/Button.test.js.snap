// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> component should match to snapshot 1`] = `
<button
  className="sc-bdVaJa fPNWws"
/>
`;

exports[`<Button/> component should modification match to snapshot 1`] = `
Array [
  <button
    className="sc-bdVaJa eNQXUi"
/>,
  <button
    className="sc-bdVaJa fnrtTC"
/>,
  <button
    className="sc-bdVaJa hlpkNP"
/>,
  <button
    className="sc-bdVaJa hnxRwy"
/>,
  <button
    className="sc-bdVaJa Iodrq"
/>,
  <button
    className="sc-bdVaJa edVvAC"
/>,
]
`;
